framework/src/postprocessors/SideIntegralFunctorPostprocessor.C

```diff
@@ -57,43 +59,75 @@ SideIntegralFunctorPostprocessorTempl<is_ad>::computeFaceInfoIntegral(const Face
   // We wont allow that case, unless explicitly requested by the user,
   // as this means the sideset is reversed, but we will allow the case where
   // both sides are defined
-  bool has_elem;
+  const bool has_elem = checkFunctorDefinedOnSideBlock();
+
+  if (has_elem)
+  {
+    Moose::SingleSidedFaceArg ssf = {
+        fi, Moose::FV::LimiterType::CentralDifference, true, false, _current_elem->subdomain_id()};
+    return MetaPhysicL::raw_value(_prefactor(ssf) * _functor(ssf));
+  }
+  else
+  {
+    if (!_partial_integral)
+      errorFunctorNotDefinedOnSideBlock();
+    return 0;
+  }
+}
+
+template <bool is_ad>
+Real
+SideIntegralFunctorPostprocessorTempl<is_ad>::computeQpIntegral()
+{
+  const bool has_elem = checkFunctorDefinedOnSideBlock();
+
+  if (has_elem)
+  {
+    Moose::ElemSideQpArg elem_side_qp = {_current_elem, _current_side, _qp, _qrule};
+    return MetaPhysicL::raw_value(_prefactor(elem_side_qp) * _functor(elem_side_qp));
+  }
+  else
+  {
+    if (!_partial_integral)
+      errorFunctorNotDefinedOnSideBlock();
+    return 0;
+  }
+}
```

Comment:
This routine is the exact same as the face info one except for the functor argument. I would consider making a template method that you call with the correct functor argument based on the virtual method being called (`computeQpIntegral` or `computeFaceInfoFaceIntegral`)

Reply:
done